build: fix tarantool build failure on xcode 12.5

EvgenyMekhanik opened the pull request:
`VERSION` files in small subproject and in tarantool are treated as C++
standard library on a filesystem with case-insensitive names. So we have
to delete the root of tarantool project from `include_directories` in
tarantool CMake. Also we have to change `include_directories` in tarantool
CMake from the root of `small` project to `small` subfolder in `small` project.
Closes #6076

sergos commented:
Please add an explanation for `crc32.c` rename in commit message. 
Also, the name selection is questionable. The src/crc32.c mentions `architecture-specific implementation`, so perhaps better choice is `crc32_impl.c`

sergos commented on src/box/alter.cc:
```diff
@@ -44,7 +44,7 @@
 #include "fiber.h" /* for gc_pool */
 #include "scoped_guard.h"
-#include "third_party/base64.h"
+#include "base64.h"
```
It looks like you wanted to update all `#include ""` for `#include <>` across the patch but some are slipped. Or is there more complex logic behind?

EvgenyMekhanik replied:
Fixed some places some places need `<>`, in other places, I fixed it by analogy in the entire patch

EvgenyMekhanik commented on CMakeLists.txt:
```diff
@@ -283,10 +283,10 @@ find_package_message(TARANTOOL_VERSION
 include_directories(${PROJECT_BINARY_DIR}/src)
 include_directories(${PROJECT_SOURCE_DIR}/src/lib)
-include_directories(${PROJECT_SOURCE_DIR}/src/lib/small)
+include_directories(${PROJECT_SOURCE_DIR}/src/lib/small/small)
```
Fixed

EvgenyMekhanik commented:
> Please add an explanation for `crc32.c` rename in commit message.
> Also, the name selection is questionable. The src/crc32.c mentions `architecture-specific implementation`, so perhaps better choice is `crc32_impl.c`

No need to changed crc32.c only crc32.h. I change crc32.c by analogy. After you comment i changed crc32.c back. And i changed crc32.h to crc32_impl.h as you recommended.

sergos commented on src/lib/core/decimal.c:
```diff
@@ -30,8 +30,8 @@
-#include "third_party/decNumber/decContext.h"
-#include "third_party/decNumber/decPacked.h"
+#include <decNumber/decContext.h>
+#include <decNumber/decPacked.h>
```
Shall we go deeper in this particular case and update include path to third_party/decNumber?

sergos commented on src/lib/core/fiber.c:
```diff
@@ -152,7 +152,7 @@ cpu_stat_end(struct cpu_stat *stat, struct clock_stat *cord_clock_stat)
 #endif /* ENABLE_FIBER_TOP */
 
-#include "third_party/valgrind/memcheck.h"
+#include <valgrind/memcheck.h>
```
ditto

sergos commented on src/lua/httpc.c:
```diff
@@ -35,7 +35,7 @@
 
 #include "http_parser/http_parser.h"
-#include "src/httpc.h"
+#include "../httpc.h"
```
wha? why that?

EvgenyMekhanik replied:
Fixed

sergos commented on src/lua/init.c:
```diff
@@ -51,8 +51,8 @@
-#include "third_party/lua-cjson/lua_cjson.h"
-#include "third_party/lua-yaml/lyaml.h"
+#include <lua-cjson/lua_cjson.h>
+#include <lua-yaml/lyaml.h>
```
same question - can we have a list of include folders rather include names with folders?

sergos commented on src/crc32.c:
```diff
@@ -31,7 +31,7 @@
 
 #include <trivia/config.h>
-#include <third_party/crc32.h>
+#include <crc32_impl.h>
```
I believe we'll have crc32_impl.h and crc32.c in third_party folder after rename only first one?
Is it consistent? For me it looks natural if header has the same name as the code file.

EvgenyMekhanik replied:
Fixed

iskander232 reviewed (approved):
LGTM

Korablev77 commented:
I tried to build on fresh Mac and I have to apply this diff to make build succeed:
```
diff --git a/third_party/lua-cjson/strbuf.c b/third_party/lua-cjson/strbuf.c
index 69057ffb3..87cdda64c 100644
--- a/third_party/lua-cjson/strbuf.c
+++ b/third_party/lua-cjson/strbuf.c
@@ -28,7 +28,7 @@
 #include <string.h>
 
 #include "strbuf.h"
-#include "small/ibuf.h"
+#include "ibuf.h"
 
 static void die(const char *fmt, ...)
 {
```

EvgenyMekhanik commented:
> I tried to build on fresh Mac and I have to apply this diff to make build succeed:
> 
> ```
> diff --git a/third_party/lua-cjson/strbuf.c b/third_party/lua-cjson/strbuf.c
> index 69057ffb3..87cdda64c 100644
> --- a/third_party/lua-cjson/strbuf.c
> +++ b/third_party/lua-cjson/strbuf.c
> @@ -28,7 +28,7 @@
>  #include <string.h>
>  
>  #include "strbuf.h"
> -#include "small/ibuf.h"
> +#include "ibuf.h"
>  
>  static void die(const char *fmt, ...)
>  {
> ```

This is a new header in this file, on the moment of may patch it was no such header)

Totktonada commented:
It would be good to reflect the new changes in the PR description (say, just copy the new commit message here).

It would be good to link the small's PR from this one.

No other objections from me.
